components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -5,3 +5,597 @@
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local CardNames = mw.loadData('Module:CardNames')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+local Streams = Lua.import('Module:Links/Stream', {requireDevIfEnabled = true})
+
+local Opponent = require('Module:OpponentLibraries').Opponent
+
+local ALLOWED_STATUSES = {'W', 'FF', 'DQ', 'L'}
+local CONVERT_STATUS_INPUT = {W = 'W', FF = 'FF', L = 'L', DQ = 'DQ', ['-'] = 'L'}
+local DEFAULT_LOSS_STATUSES = {'FF', 'L', 'DQ'}
+local MAX_NUM_OPPONENTS = 2
+local DEFAULT_BEST_OF = 99
+local EPOCH_TIME_EXTENDED = '1970-01-01T00:00:00+00:00'
+local TODAY = os.date('%Y-%m-%d')
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(match)
+	Table.mergeInto(
+		match,
+		CustomMatchGroupInput._readDate(match)
+	)
+	match = CustomMatchGroupInput._getExtraData(match)
+	match = CustomMatchGroupInput._getTournamentVars(match)
+	match = CustomMatchGroupInput._checkFinished(match)
+	match = CustomMatchGroupInput._adjustData(match)
+	match = CustomMatchGroupInput._getVodStuff(match)
+	match = CustomMatchGroupInput._getLinks(match)
+
+	return match
+end
+
+function CustomMatchGroupInput._readDate(matchArgs)
+	if matchArgs.date then
+		return MatchGroupInput.readDate(matchArgs.date)
+	else
+		return {
+			date = EPOCH_TIME_EXTENDED,
+			dateexact = false,
+			timestamp = DateExt.epochZero,
+		}
+	end
+end
+
+function CustomMatchGroupInput._checkFinished(match)
+	if Logic.readBoolOrNil(match.finished) == false then
+		match.finished = false
+	elseif Logic.readBool(match.finished) then
+		match.finished = true
+	elseif Logic.isNotEmpty(match.winner) then
+		match.finished = true
+	end
+
+	-- Match is automatically marked finished upon page edit after a
+	-- certain amount of time (depending on whether the date is exact)
+	if match.finished ~= true and match.timestamp > DateExt.epochZero then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local threshold = match.dateexact and 30800 or 86400
+		if match.timestamp + threshold < currentUnixTime then
+			match.finished = true
+		end
+	end
+
+	return match
+end
+
+function CustomMatchGroupInput._getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'solo'))
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function CustomMatchGroupInput._getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod)
+
+	return match
+end
+
+function CustomMatchGroupInput._getLinks(match)
+	match.links = {
+		preview = match.preview,
+		preview2 = match.preview2,
+		interview = match.interview,
+		interview2 = match.interview2,
+		review = match.review,
+		recap = match.recap,
+	}
```

Comment:
**from #2253 review:**


__Rathoz:__
> Preview is never used
Preview2 is never used
Interview is never used
Interview2 is never used
Review is never used
Recap is never used

I'd it goes away. If it turns out to be needed, it's added then

__hjp:__
> so kick all links?
(i do not really care but vishu the cr editor said he wants to use them)

Reply:
Yes let's remove them and add them if they are actually used